### PR TITLE
Remove class `SingletonFamily` and MF attribute `id`

### PR DESCRIPTION
**Pull request by CunliangGeng**
Singleton molecular families can be well represented with the class `MolecularFamily`, and it's not needed to keep a specific class `SingletonFamily` for them, so this PR removed it.

As for the family id, the class `SingleFamily` automatically set it (i.e.`"singleton-" + uuid`), while the class `MolecularFamily` requires users to provide the family id during initiation. It makes more sense to let users or loaders of molecular family to decide how to set family id. So in the `GNPSMolecularFamilyLoader` class, the name of singleton MF was updated to `"singleton-" + spectrum_id`.

The attribute `id` of MolecularFamily is unused by anywhere, so it's removed too.

### CODE REVIEW

**Comment by CunliangGeng** [comment]
Current dependencies on/for this PR:
* `dev`
  * **PR #178** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/178?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #181** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/181?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #182** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/182?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #183** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/183?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #184** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/184?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #185** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/185?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #186** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/186?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #187** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/187?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #188** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/188?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #189** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/189?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                      * **PR #190** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/190?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                        * **PR #191** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/191?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                          * **PR #192** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/192?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
                            * **PR #193** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/193?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                              * **PR #194** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/194?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                * **PR #196** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/196?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                  * **PR #197** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/197?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                    * **PR #198** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/198?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                      * **PR #199** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/199?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                        * **PR #200** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/200?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                          * **PR #201** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/201?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                            * **PR #202** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/202?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                * **PR #195** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/195?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/192?utm_source=stack-comment).

**Comment by CunliangGeng** [comment]
Current dependencies on/for this PR:
* dev
  * **PR #178** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/178?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #181** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/181?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #182** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/182?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #183** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/183?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #184** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/184?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #185** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/185?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #186** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/186?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #187** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/187?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #188** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/188?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #189** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/189?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                      * **PR #190** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/190?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                        * **PR #191** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/191?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                          * **PR #192** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/192?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/192?utm_source=stack-comment).

File: src/nplinker/metabolomics/molecular_family.py [hunk 1]
```diff
@@ -13,14 +13,12 @@ def __init__(self, family_id: str):
         """Class to model molecular family.
 
         Args:
-            family_id(str): Id for the molecular family.
+            family_id(str): Unique id for the molecular family.
```
Comment (gcroci2) [inline src/nplinker/metabolomics/molecular_family.py:16]: I know that args and attributes can be and are different, but I think that the best practice is that args of the init method are always then made attributes of the class. So in general I'd avoid using "Args" in the doc string, and I'd stick to "Attributes" only. This way there is less redundancy. But I noticed this also in other PRs, so maybe it's just easier to keep the redundancy, which states something correct anyway.

File: src/nplinker/metabolomics/gnps/gnps_molecular_family_loader.py [hunk 2]
```diff
@@ -19,6 +18,13 @@ def __init__(self, file: str | PathLike):
         3. FEATURE-BASED-MOLECULAR-NETWORKING
             - networkedges_selfloop/*.selfloop
 
+        The "ComponentIndex" column in the GNPS molecular family file is treated
+        as family id. But for molecular families that have only one member (i.e. spectrum),
+        which are called singleton molecular family, they have the same value of
```
Comment (gcroci2) [inline src/nplinker/metabolomics/gnps/gnps_molecular_family_loader.py]:
```suggestion
        named singleton molecular families, have the same value of
```

File: src/nplinker/metabolomics/gnps/gnps_molecular_family_loader.py [hunk 3]
```diff
@@ -19,6 +18,13 @@ def __init__(self, file: str | PathLike):
         3. FEATURE-BASED-MOLECULAR-NETWORKING
             - networkedges_selfloop/*.selfloop
 
+        The "ComponentIndex" column in the GNPS molecular family file is treated
```
Comment (gcroci2) [inline src/nplinker/metabolomics/gnps/gnps_molecular_family_loader.py]:
```suggestion
        The "ComponentIndex" column in the GNPS molecular family's file is treated
```

File: src/nplinker/metabolomics/gnps/gnps_molecular_family_loader.py [hunk 4]
```diff
@@ -19,6 +18,13 @@ def __init__(self, file: str | PathLike):
...
+        as family id. But for molecular families that have only one member (i.e. spectrum),
```
Comment (gcroci2) [inline src/nplinker/metabolomics/gnps/gnps_molecular_family_loader.py:22]:
```suggestion
        as family id. But all the molecular families that have only one member (i.e. spectrum),
```

File: src/nplinker/metabolomics/gnps/gnps_molecular_family_loader.py (hunk 2 above)
Comment (gcroci2) [inline src/nplinker/metabolomics/gnps/gnps_molecular_family_loader.py:23]: About "they have the same value of", please specify which field you're referring to (I think family id)

**Review by gcroci2 — APPROVED** [review]

File: src/nplinker/metabolomics/molecular_family.py (hunk 1 above)
Comment (CunliangGeng) [reply src/nplinker/metabolomics/molecular_family.py:16]: You're right that `Args` and `Attributes` serve different purposes. Better to keep both, especially when there are attributes that are not generated from `Args`. Clarity is over redundancy in this case, I would say ;-)

File: src/nplinker/metabolomics/gnps/gnps_molecular_family_loader.py (hunk 4 above)
Comment (CunliangGeng) [reply src/nplinker/metabolomics/gnps/gnps_molecular_family_loader.py:22]: Updated in a different way.

**Comment by CunliangGeng** [comment]
### Merge activity

* **Dec 19, 8:24 AM**: @@CunliangGeng started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/192).
* **Dec 19, 8:27 AM**: [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/192) rebased this pull request as part of a merge.
* **Dec 19, 8:28 AM**: @@CunliangGeng merged this pull request with [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/192).

**Pull request merged**